### PR TITLE
Cloud Docs

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

- [x] Default to `cloud.appwrite.io` in examples -> Related: https://github.com/appwrite/sdk-generator/pull/646
- [x] Rewrite Install -> Self-hosting as a complete self-hosting guide
- [x] Move all environment variables to self-hosting https://github.com/appwrite/appwrite/pull/5394
- [x] Add public cloud beta instructions.

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/views/docs/configuration.phtml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<p>
+    Self-hosting allows you to customize Appwrite to meet your needs and preferences. Here's a list of things you will need to setup before developing with Appwrite. Before you start, backup your <code>.env</code> and <code>docker-compose.yml</code> files.
+</p>
+
+<h2><a href="#localhost" id="localhost">Localhost Development</a></h2>
+<p>Self-hosting Appwrite for development lets you quickly prototype on your local machine. Since localhost is only a loopback address that points back at your development machine, there are some caveats you should be aware of when working with localhost.</p>
+
+[Self-signed stuff. Describe what the selfsigned method does, why you need it, and when to use it.]
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:8]: @PineappleIOnic

File: app/views/docs/configuration.phtml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+[Proxy information for android: What should they know? How do they access local host on android?]
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:9]: @PineappleIOnic

File: app/views/docs/configuration.phtml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+<p>If you're testing your Flutter, Android, or iOS app on a mobile device, you will not be able to access the localhost of your PC or Mac. When testing with a physical device, you may use a service like <a href="https://ngrok.com/" target="_blank" rel="noopener">ngrok</a> to proxy the Appwrite instance.</p>
+[cookies? What's weird about cookies while developing on local host?]
```
Comment (gewenyu99) [inline app/views/docs/configuration.phtml:11]: @PineappleIOnic

File: app/views/docs/functions.phtml [hunk 4]
```diff
@@ -781,7 +779,7 @@ func main() async throws {
     </tbody>
 </table>
 
-<p>By default, the following runtimes are enabled: <code>node-16.0</code>, <code>php-8.0</code>, <code>python-3.9</code>, <code>ruby-3.0</code>. To enable or disable runtimes, you can edit the <code>_APP_FUNCTIONS_RUNTIMES</code> <a href="/docs/environment-variables#functions">environment variable</a>.</p>
+<p>By default, the following runtimes are enabled: <code>node-16.0</code>, <code>php-8.0</code>, <code>python-3.9</code>, <code>ruby-3.0</code>.</p>
```
Comment (christyjacob4) [inline app/views/docs/functions.phtml:782]: node-16.0, php-8.0, python-3.9, ruby-3.0, dart-2.17

File: app/views/docs/authentication-server.phtml [hunk 5]
```diff
@@ -25,8 +25,8 @@
             <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Client, Account } from "appwrite";
 
 const client = new Client()
-    .setEndpoint('https://[HOSTNAME_OR_IP]/v1') // Your API Endpoint
-    .setProject('5df5acd0d48c2');               // Your project ID
+    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject('[PROJECT_ID]');               // Your project ID
```
Comment (christyjacob4) [inline app/views/docs/authentication-server.phtml:29]: Based on this convention https://stackoverflow.com/questions/21503865/how-to-denote-that-a-command-line-argument-is-optional-when-printing-usage

Required arguments are denoted by angled brackets `<required_argument>` and optional ones as `[optional_argument]` . So I think we should follow this widely used convention as well .

File: app/views/docs/rest.phtml [hunk 6]
```diff
@@ -78,7 +78,7 @@ Set-Cookie: a_session_61e71ec784ab035f7259=eyJ0...aSJ9; expires=Tue, 19-Dec-2023
     <pre class="line-numbers"><code class="prism language-http" data-prism>GET /v1/account HTTP/1.1
 Cookie: a_session_61e71ec784ab035f7259_legacy=eyJ0...aSJ9; a_session_61e71ec784ab035f7259=eyJ0...aSJ9
 Content-Type: application/json
-X-Appwrite-Project: 5df5acd0d48c2</code></pre>
```
Comment (christyjacob4) [inline app/views/docs/rest.phtml:81]: Might wanna replace references to `demo.appwrite.io` in L72 and L73

File: app/views/docs/production.phtml [hunk 7]
```diff
@@ -27,7 +27,11 @@
...
 
-<h2><a href="/docs/production#emails" id="emails">Emails</a></h2>
+<h2><a href="/docs/production#disable-limits" id="disable-limits">Enable Rate Limits</a></h2>
+<p>If you disabled rate limits during development, make sure you re-enable them when moving to production environments. Rate limiting can be enabled by setting the <span class="tag">_APP_OPTIONS_ABUSE</span> environment variable to <code>enabled</code>.</p>
+<p>
+<a href="/docs/environment-variables"><i class="icon-angle-circled-right"></i>Learn more about environment variables</a>
+</p>
 
 <p>Sending emails is hard. There are a lot of SPAM rules and configurations to master in order to set a functional SMTP server. The SMTP server that comes packaged with Appwrite is great for development but needs some work done to function well against SPAM filters.</p>
```
Comment (christyjacob4) [inline app/views/docs/production.phtml:36]: You've missed out the <a> tag to denote the beginning of the emails section

File: app/views/docs/main.phtml [hunk 8]
```diff
@@ -92,11 +97,10 @@ $getSpecsUrl = function($type, $platform) use ($latestVersion) {
     </div>
 
     <div class="margin-bottom-large">
-        <h2 class="margin-bottom">API References</h2>
-        
-        <p>The Appwrite API is organized around REST. Our API has predictable resource-oriented URLs, accepts form-encoded or JSON-encoded request bodies, returns JSON-encoded responses, and uses standard HTTP response codes, authentication, and verbs.</p>
-
-        <p>Specifications for the Appwrite API are available in multiple formats, including Open API 3 for <a href="<?php echo $getSpecsUrl('open-api3', 'client'); ?>">clients</a> or <a href="<?php echo $getSpecsUrl('open-api3', 'server'); ?>">servers</a> and Swagger2 for <a href="<?php echo $getSpecsUrl('swagger2', 'client'); ?>">clients</a> or <a href="<?php echo $getSpecsUrl('swagger2', 'server'); ?>">servers</a>.</p>
+        <h2 class="margin-bottom">Explore Appwrite</h2>
+        <p>Appwrite provides all the core features of a model backend. Explore Appwrite's features for <a href="/docs/authentication">authentication</a>, <a href="/docs/databases">databases</a>, <a href="/docs/storage">file storage</a>, and executing server side code with <a href="/docs/functions">functions</a>.</p>
```
Comment (christyjacob4) [inline app/views/docs/main.phtml:101]: model backend?

File: app/views/docs/configuration.phtml [hunk 9]
```diff
@@ -0,0 +1,65 @@
+<p>
+    Self-hosting allows you to customize Appwrite to meet your needs and preferences. Here's a list of things you will need to setup before developing with Appwrite. Before you start, backup your <code>.env</code> and <code>docker-compose.yml</code> files.
+</p>
+
+<h2><a href="#localhost" id="localhost">Localhost Development</a></h2>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:5]: Localhost development ? Rather Local Development ?

File: app/views/docs/configuration.phtml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+<p>Self-hosting Appwrite for development lets you quickly prototype on your local machine. Since localhost is only a loopback address that points back at your development machine, there are some caveats you should be aware of when working with localhost.</p>
+<p>Appwrite generates a self-signed certificate for iOS and Android development on localhost, as they don't allow unsecure HTTP by default. If you're developing Android or iOS apps on localhost, set <code>setSelfSigned(true)</code> while initializing your Appwrite client.</p>
+<p>If you're using an Android emulator to test your Flutter or Android apps, you can access localhost of your machine using the special alias IP <code>10.0.2.2</code>.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:8]: Im not sure 10.0.2.2 is true for all cases

File: app/views/docs/configuration.phtml [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+<p>If you're testing your Flutter, Android, or iOS app on a mobile device, you will not be able to access the localhost of your PC or Mac. When testing with a physical device, you may use a service like <a href="https://ngrok.com/" target="_blank" rel="noopener">ngrok</a> to proxy the Appwrite instance.</p>
+
+<h2><a href="#delivery" id="delivery">Setup Email and SMS Delivery</a></h2>
+<p>
+    Appwrite requires an SMTP service to deliver emails. Email delivery is required for features like password recovery, account verification, MagicURL authentication, and team invites. You can choose to a third-party SMTP service or hosting your own SMTP server.
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:13]:
```suggestion
    Appwrite requires an SMTP service to deliver emails. Email delivery is required for features like password recovery, account verification, MagicURL authentication, and team invites. You can use a third-party SMTP service or host your own SMTP server.
```

File: app/views/docs/configuration.phtml [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+</p>
+<p><a href="/docs/email-delivery">Set up email delivery</a></p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:15]: Does this page exist?

File: app/views/docs/configuration.phtml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+<p>
+    Appwrite requires a third-party SMS delivery service to send SMS messages. SMS delivery is required for phone authentication.
+</p>
+<p><a href="/docs/email-delivery">Set up SMS delivery</a></p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:20]: Does this page exist? Also it should be `sms-delivery`

File: app/views/docs/configuration.phtml [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+<h2><a href="#storage" id="storage">Storage Adaptors</a></h2>
+<p>Appwrite's Storage Service can be configured to store files locally, or self-hosted and cloud storage services. By default, Appwrite's Storage Service <b>stores files on your server's local storage</b>. If you expect large volumes of data or the need to have scalable data storage, you may choose to use a separate storage service.</p>
+
+<p>Appwrite supports AWS S3, Digital Ocean Spaces, Backblazk, Linode, and Wasabi as storage adaptors. Some of these services can be self-hosted, just like Appwrite.</p>
+
+<p>You can select which storage adaptor to use by setting the <code>_APP_STORAGE_DEVICE</code> environment variable. Valid values are 'local', 's3', 'dospaces', 'backblaze', 'linode' and 'wasabi'. Each storage adaptor requires their own set of environment variables to configure.</p>
+
+<p>The maximum file size allowed for file upload is controlled by the <code>_APP_STORAGE_LIMIT</code> environment variable, which defaults to 30 MB. See [Environment Variables](/docs/environment-variables#storage) for more information.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:47]: This is using markdown syntax. It needs to be updated to use an <a> tag

File: app/views/docs/databases.phtml [hunk 15]
```diff
@@ -398,4 +398,501 @@ func main() async throws{
 }</code></pre>
         </div>
     </li>
-</ul>
\ No newline at end of file
+</ul>
+
+<p>The following query methods are currently supported:</p>
```
Comment (christyjacob4) [inline app/views/docs/databases.phtml:403]: Ah why is this change here? I think its not synced with master yet

File: app/views/docs/self-hosting.phtml [hunk 16]
```diff
@@ -1,4 +1,4 @@
-<p>Appwrite is a self-hosted backend server, packaged as a set of Docker containers. You can install and run Appwrite on any operating system that can run a <a href="https://www.docker.com/products/docker-desktop" target="_blank" rel="noopener">Docker CLI</a>. You can use Appwrite on your local machine or a cloud provider of your choice.</p>
+<p>Appwrite can be easily self-hosted on your local machine or a server. You can install and run Appwrite on any operating system that can run a <a href="https://www.docker.com/products/docker-desktop" target="_blank" rel="noopener">Docker CLI</a>. Self-hosted Appwrite instances can be configured flexibly with access to the same features found on Appwrite Cloud.</p>
```
Comment (christyjacob4) [inline app/views/docs/self-hosting.phtml:1]: let's rephrase it and say something along the lines of "Appwrite was designed from the ground up with self-hosting in mind"

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

File: app/views/docs/authentication-server.phtml (hunk 5 above)
Comment (gewenyu99) [reply app/views/docs/authentication-server.phtml:29]: YES! but also this is why we need to move to markdown!!!!

We need to escape it every time because it's taken as an HTML tag!

File: app/views/docs/configuration.phtml (hunk 10 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:8]: https://developer.android.com/studio/run/emulator-networking

Found it here. I think it's meant to be a feature for android emulators

File: app/views/docs/configuration.phtml (hunk 12 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:15]: Yes, https://appwrite.io/docs/email-delivery

File: app/views/docs/configuration.phtml (hunk 13 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:20]: Ooops

File: app/views/docs/main.phtml (hunk 8 above)
Comment (gewenyu99) [reply app/views/docs/main.phtml:101]: Modern. hja

File: app/views/docs/production.phtml (hunk 7 above)
Comment (gewenyu99) [reply app/views/docs/production.phtml:36]: You're right I need to update what's written here anyway.

File: app/views/docs/self-hosting.phtml (hunk 16 above)
Comment (gewenyu99) [reply app/views/docs/self-hosting.phtml:1]: Why don't we just say Appwrite was designed from the ground up with self-hosting in mind, I like it XD

**Comment by gewenyu99** [comment]
gewenyu99-laughing-space-tribble-96g4qw464g5f99j4-2080 preview app github dev_docs
Preview for the main page.

**Comment by gewenyu99** [comment]
gewenyu99-laughing-space-tribble-96g4qw464g5f99j4-2080 preview app github dev_docs_self-hosting

**Comment by gewenyu99** [comment]
gewenyu99-laughing-space-tribble-96g4qw464g5f99j4-2080 preview app github dev_docs_configuration

File: app/views/docs/configuration.phtml (hunk 3 above)
Comment (stnguyen90) [reply app/views/docs/configuration.phtml]: So....these 2 things aren't actually exclusive to self-hosted. Feel free to tweak accordingly.

### Cookies

Appwrite primarily uses cookies to manage sessions. This works great in most scenarios on the web because cookies are automatically handled. However, there are some cases where they may not work correctly. 

#### Third-Party Cookies

Although cookies are typically automatically sent to the server based on the domain, some browsers won't include the cookie if the cookie is considered a 3rd party cookie. To make Appwrite's cookie a 1st party cookie, you'll need create a [Custom Domain](https://appwrite.io/docs/custom-domains) for your project and use that domain for your endpoint.

#### Server-Side Rendering (SSR)

Many frameworks (e.g. NextJS, Nuxt, SevelteKit) include SSR, a technique in which code is executed server-side instead of client-side. If a session is created server-side, the cookies are not persisted, and subsequent API calls will be unauthenticated. Similarly, if a session is created client-side, the cookie will not be available when API calls are made server-side.

File: app/views/docs/configuration.phtml (hunk 10 above)
Comment (christyjacob4) [reply app/views/docs/configuration.phtml:8]: Okay then 👍  sounds good

File: app/views/docs/configuration.phtml [hunk 17]
```diff
@@ -0,0 +1,65 @@
+<p>
+    Self-hosting allows you to customize Appwrite to meet your needs and preferences. Here's a list of things you will need to set up before developing with Appwrite. Before you start, back up your <code>.env</code> and <code>docker-compose.yml</code> files.
+</p>
+
+<h2><a href="#local-development" id="local-development">Local Development</a></h2>
+<p>Self-hosting Appwrite for development lets you quickly prototype on your local machine. Since localhost is only a loopback address that points back at your development machine, there are some caveats you should be aware of when working with localhost.</p>
+<p>Appwrite generates a self-signed certificate for iOS and Android development on localhost, as they don't allow unsecured HTTP by default. If you're developing Android or iOS apps on localhost, set <code>setSelfSigned(true)</code> while initializing your Appwrite client.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:7]: I dont think its iOS and Android specific. it applies to flutter and web as well

File: app/views/docs/configuration.phtml [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+<p>If you're testing your Flutter, Android, or iOS app on a mobile device, you will not be able to access the localhost of your PC or Mac. When testing with a physical device, you may use a service like <a href="https://ngrok.com/" target="_blank" rel="noopener">ngrok</a> to proxy the Appwrite instance.</p>
+
+<h2><a href="#delivery" id="delivery">Setup Email and SMS Delivery</a></h2>
+<p>
+    Appwrite requires an SMTP service to deliver emails. Email delivery is required for features like password recovery, account verification, MagicURL authentication, and team invites. You can use a third-party SMTP service or host your own SMTP server.
+</p>
+<p><a href="/docs/email-delivery">Set up email delivery</a></p>
+
+<p>
+    Appwrite requires a third-party SMS delivery service to send SMS messages. SMS delivery is required for phone authentication.
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:18]: Can we rephrase it to sound more like a feature than a limitation ? 

`Appwrite supports multiple third-party SMS delivery providers .. ( flexibility, no-vendor locks etc.  )`

File: app/views/docs/configuration.phtml [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+
+<h2><a href="#storage" id="storage">Storage Adaptors</a></h2>
+<p>Appwrite's Storage Service can be configured to store files locally, or with self-hosted and cloud storage services. By default, Appwrite's Storage Service <b>stores files on your server's local storage</b>. If you expect large volumes of data or the need to have scalable data storage, you may choose to use a separate storage service.</p>
+
+<p>Appwrite supports AWS S3, Digital Ocean Spaces, Backblaze, Linode, and Wasabi as storage adaptors. Some of these services can be self-hosted, just like Appwrite.</p>
+
+<p>You can select which storage adaptor to use by setting the <code>_APP_STORAGE_DEVICE</code> environment variable. Valid values are <code>local</code>, <code>s3</code>, <code></code>, <code>backblaze</code>, <code>linode</code> and <code></code>. Each storage adaptor requires its own set of environment variables to configure.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:45]: dospaces, wasabi missing

File: app/views/docs/configuration.phtml [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+
+<p>The maximum file size allowed for file upload is controlled by the <code>_APP_STORAGE_LIMIT</code> environment variable, which defaults to 30 MB. See <a href="/docs/environment-variables#storage">Environment Variables</a> for more information.</p>
```
Comment (christyjacob4) [inline app/views/docs/configuration.phtml:47]: Can we be more clear and mention this is "per file" size limit ? and not to the entirety of the storage service?

File: app/views/docs/main.phtml [hunk 21]
```diff
@@ -57,15 +57,20 @@ $getSpecsUrl = function($type, $platform) use ($latestVersion) {
...
+    </div>
+
     <h2 class="margin-bottom">Overview</h2>
 
-    <p>Appwrite is an end-to-end backend server that aims to abstract the complexity of common, complex, and repetitive tasks required for building a modern app.</p>
+    <p>Appwrite is a secure, open-source backend that abstracts the complexity of common, complex, and repetitive tasks required for building a modern app.</p>
 
-    <p>Appwrite provides you with a set of APIs, tools, and a management console UI to help you build your apps a lot faster, and in a much more secure way. Within Appwrite's different services, you can find user authentication and account management, user preferences, database and storage persistence, cloud functions, localization, image manipulation, and more.</p>
+    <p>Appwrite provides you with a set of APIs, tools, and a management console UI to help you build your apps quickly without compromising security. Appwrite includes features like user authentication and account management, user preferences, database and file storage, cloud functions, localization, image manipulation, and more.</p>
```
Comment (christyjacob4) [inline app/views/docs/main.phtml:71]: Can we use some other word other than security ? This former part of the sentence does not showcase how security could be compromised

File: app/views/docs/production.phtml [hunk 22]
```diff
@@ -27,11 +27,15 @@
...
+<p>
+<a href="/docs/environment-variables"><i class="icon-angle-circled-right"></i>Learn more about environment variables</a>
+</p>
 
-<p>Another easier option is to use an 'SMTP as a service' product like Sendgrid or Mailgun. You can <a href="/docs/email-delivery">change Appwrite SMTP settings</a> and credentials to any 3rd party provider you like who support SMTP integration using our Docker environment variables. Most SMTP providers offer a decent free tier to get started with.</p>
+<h2><a href="/docs/production#emails" id="emails">Emails</a></h2>
+<p>Sending emails is hard. There are a lot of spam rules and configurations to master in order to set up a functional SMTP server. While it is okay to use a self-hosted SMTP server during development, you should use a third-party SMTP provider for production so your email doesn't get labeled as spam.</p>
+<p>You can <a href="/docs/email-delivery">change Appwrite's SMTP settings</a> and credentials to any 3rd party provider you like who supports SMTP integration using our Docker environment variables. Most SMTP providers offer a decent free tier to get started with.</p>
```
Comment (christyjacob4) [inline app/views/docs/production.phtml:38]:
```suggestion
<p>You can <a href="/docs/email-delivery">change Appwrite's SMTP settings</a> and credentials to any 3rd party provider you like that supports SMTP integration using our Docker environment variables. Most SMTP providers offer a decent free tier to get started with.</p>
```

File: app/views/docs/main.phtml (hunk 8 above)
Comment (christyjacob4) [reply app/views/docs/main.phtml]: 😆

File: app/views/docs/self-hosting.phtml (hunk 16 above)
Comment (christyjacob4) [reply app/views/docs/self-hosting.phtml]: Sounds good 👌

File: app/views/docs/databases.phtml (hunk 15 above)
Comment (christyjacob4) [reply app/views/docs/databases.phtml:403]: This is still not resolved ?

Comment (christyjacob4) [reply app/views/docs/databases.phtml:403]: It has been moved to https://github.com/appwrite/docs/blob/main/app/views/docs/queries.phtml

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Awesome work 👌  Final set of comments. Please also check the unresolved comment from the previous review

File: app/views/docs/main.phtml (hunk 21 above)
Comment (gewenyu99) [reply app/views/docs/main.phtml:71]: No, it's that you can build **quickly** and still not compromise on **security**. One less thing to worry about. I think security is the only word here as well.

File: app/views/docs/configuration.phtml (hunk 20 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:47]: Done

File: app/views/docs/configuration.phtml (hunk 19 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:45]: Oops

File: app/views/docs/configuration.phtml (hunk 18 above)
Comment (gewenyu99) [reply app/views/docs/configuration.phtml:18]: I like this.

File: app/views/docs/databases.phtml (hunk 15 above)
Comment (gewenyu99) [reply app/views/docs/databases.phtml:403]: Merge conflit probably :P This branch is ancient.

**Review by christyjacob4 — APPROVED** [review]
